File: billing-application/src/main/java/com/example/billing/application/port/in/CreateBudgetAlertRuleUseCase.java
package com.example.billing.application.port.in;

import com.example.billing.application.command.CreateBudgetAlertRuleCommand;
import com.example.billing.domain.budget.BudgetAlertRule;

public interface CreateBudgetAlertRuleUseCase {
    BudgetAlertRule create(CreateBudgetAlertRuleCommand command);
}

File: billing-application/src/main/kotlin/com/example/billing/application/port/in/CreateBudgetAlertRuleUseCase.kt
package com.example.billing.application.port.`in`

import com.example.billing.application.command.CreateBudgetAlertRuleCommand
import com.example.billing.domain.budget.BudgetAlertRule

interface CreateBudgetAlertRuleUseCase {
    fun create(command: CreateBudgetAlertRuleCommand): BudgetAlertRule
}
